### ISSUE TITLE
Presentation Details not updated correctly after using Add-Rendering in script

### ISSUE THREAD

**Issue by BillCacy** [SitecorePowerShell/Console#906]
Recently i was troubleshooting an issue with renderings not being displayed in the Presentation Details of an item after running a script that used the Add-Rendering command.
I reached out to Sitecore support for assistance and they provided the following information. The version of  PSE used was 4.1, and was also reproducible using PSE 4.5.

Please let me know if you need any additional information regarding this issue.

Below is the excerpt from Sitecore support (ticket 486120).

I did a deep an extensive investigation.
Here is what I have found:
The 'Add-Rendering' PowerShell command is implemented by the AddRenderingCommand class of the Sitecore PowerShell shared source module.
It uses the ProcessLayout(Item item, LayoutDefinition layout, DeviceDefinition device) method to add a rendering to the layout.
The layout value comes from the BaseLayoutPerDeviceCommand class.
It uses the following logic:
LayoutField layoutField = item.Fields[LayoutFieldId];
var layout = LayoutDefinition.Parse(layoutField.Value);
ProcessLayout(item, layout, device);

The LayoutFieldId is set by the BaseLayoutCommand class:
if (FinalLayout) //i.e. if the -FinalLayout flag is added
  LayoutFieldId = Sitecore.FieldIDs.FinalLayoutField

layoutField.Value property is implemented by Sitecore and uses the Sitecore.Data.Fields.LayoutField.GetFieldValue(Field field) method, where filed is a LayoutField. In our case it is the '__Final Renderings' field because of the '-FinalLayout' flag.
According to Sitecore documentation,
Prior to Sitecore 8, the public static string GetFieldValue([NotNull] Field field) method only returned layout data from the shared __Renderings field. In Sitecore 8 and later, this method can return layout data either from the shared __Renderings field or from the versioned __Final Renderings field.

The method does not return the raw value of the field. It uses the layout patching workflow to calculate the resulting layout, and it returns this layout.

The issue comes from the fact that the ProcessLayout method does not save the value properly.
It writes the resulting layout to the '__Final Renderings' field instead of saving the delta only.
It does:
item.Edit(p =>
            {
                var outputXml = layout.ToXml();
                Item[LayoutFieldId] = outputXml;
            });

Instead, it should do the following:
item.Edit(p =>
            {
Sitecore.Data.Fields.LayoutField.SetFieldValue(layoutField, layout.ToXml());
            });

To sum up, the issue is caused by a bug in the Sitecore PowerShell shared source module.
Please report this issue as a bug in the module using the issue tarcking page:
https://github.com/SitecorePowerShell/Console/issues

You can test the sample code that I mentioned above and create a pull request to fix the issue in the module.

To resolve the issue you observe please perform the following:
1. Back up the master database.
2. Execute the following code for all affected items:
LayoutField layoutField = item.Fields[Sitecore.FieldIDs.FinalLayoutField];
var layout = LayoutDefinition.Parse(layoutField.Value);
Sitecore.Data.Fields.LayoutField.SetFieldValue(layoutField, layout.ToXml());

**Comment by AdamNaj** [comment]
Oh! I've updated the Set-Rendering but never updated the other Cmdlets!
Good timing! Just before I managed to release 4.6!

**Comment by AdamNaj** [comment]
Fixed.
